audio/descriptors.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2017 Kyle Robbertze
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
```

Comment:
Licensing is complicated -- here is a hugely long reply that I had hope you wouldn't have to understand / deal with :-P

Couple of things;
 * If this is based on another file, then the copyright at the top with just your name isn't correct.
 * There are actually two COPYING files in this repo. `COPYING` (GPLv3) and `COPYING.LESSER` (LGPLv3). Many files however have copyright headers which claim `LGPL V2.1 or later` clauses at the top. Hence, you should check the license at the top of the file, rather than the `COPYING` at the top level.
 * Almost all of the fx2lib code (under include/lib) is under a `LGPL V2.1 or later` clause. Most of the examples are also under `LGPL V2.1 or later` clause but some are under a `GPL V3.0 or later` clause.
 * The bulkloop example is under `LGPL V2.1 or later` see - https://github.com/mithro/fx2lib/tree/master/examples/bulkloop and https://github.com/mithro/fx2lib/blob/linux-descriptors/examples/bulkloop/bulkloop.c

Why is this important? Because Linux is licensed under `GPLv2` -- **without** an `or later` clause. This causes it to be incompatible with interacting with `GPLv3` code and you can't distribute something which includes both `GPLv3 in any form` and `GPLv2 only` as a single result. See this diagram;
![image](https://user-images.githubusercontent.com/21212/27773960-c6435606-5fc9-11e7-9b93-878e15232595.png)

Thus, to use the linux headers we need to make sure we are only using code which is `LGPLv2.1 or later`, `GPLv2 or later`, `GPLv2`, or `LGPLv2.1`. If you finding cases where that is not happening, we should fix it.

Lastly, Makefiles which are GPLv3.0 are okay because they don't end up being included in the final output. We should still however we should try and remove / replace them if possible. Ultimately, I would say that it would be pretty hard for someone to make the argument that we are not working within the spirit of the licenses but that obviously isn't really something that hardliners like Debian would be all that happy with until properly fixed.